src/input/yaml.rs

```diff
@@ -0,0 +1,211 @@
+//! YAML input processing functionality for Sprocket.
+
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::bail;
+use tracing::debug;
+
+/// Identifies the format of an input file.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum InputFormat {
+    /// JSON format
+    Json,
+    /// YAML format
+    Yaml,
+}
+
+/// Parses an input file as either JSON or YAML.
+///
+/// Tries to parse as JSON first, then YAML if JSON parsing fails.
+/// Returns the parsed data as a JSON value along with the detected format.
+pub fn parse_input_file(path: &PathBuf) -> Result<(InputFormat, serde_json::Value)> {
+    let content = fs::read_to_string(path)
+        .with_context(|| format!("Failed to read file: {}", path.display()))?;
+
+    // Try parsing as JSON first
+    if let Ok(json_value) = serde_json::from_str::<serde_json::Value>(&content) {
+        debug!("File parsed as JSON: {}", path.display());
+        return Ok((InputFormat::Json, json_value));
+    }
+
+    // Then try parsing as YAML
+    if let Ok(yaml_value) = serde_yaml::from_str::<serde_yaml::Value>(&content) {
+        debug!("File parsed as YAML: {}", path.display());
+        // Convert YAML to JSON
+        let json_value = serde_json::to_value(&yaml_value)
+            .context("Failed to convert YAML value to JSON value")?;
+        return Ok((InputFormat::Yaml, json_value));
+    }
+
+    // If neither worked, return an error
+    bail!(
+        "File is neither valid JSON nor valid YAML: {}",
+        path.display()
+    )
+}
+
+/// Gets a JSON string representation of the input file.
+pub fn get_json_string(input_path: PathBuf) -> Result<String> {
+    let (format, json_value) = parse_input_file(&input_path)?;
+
+    match format {
+        InputFormat::Json => {
+            debug!("Using JSON input directly");
+            serde_json::to_string_pretty(&json_value)
+                .context("Failed to convert JSON value to string")
+        }
+        InputFormat::Yaml => {
+            debug!("Using YAML input converted to JSON");
+            serde_json::to_string_pretty(&json_value)
+                .context("Failed to convert YAML-derived JSON value to string")
+        }
+    }
+}
```

Comment:
This isn't being used, correct? Please delete it. We don't like to have "dead code" but because you made this `pub` it isn't being flagged.